src/main/java/de/tum/in/www1/artemis/service/QuizExerciseService.java

```diff
@@ -480,4 +201,50 @@ private void resetInvalidQuestions(QuizExercise quizExercise) {
             question.setInvalid(false);
         }
     }
+
+    @Override
+    public QuizExercise save(QuizExercise quizExercise) {
+        quizExercise.setMaxPoints(quizExercise.getOverallQuizPoints());
+
+        // create a quizPointStatistic if it does not yet exist
+        if (quizExercise.getQuizPointStatistic() == null) {
+            var quizPointStatistic = new QuizPointStatistic();
+            quizExercise.setQuizPointStatistic(quizPointStatistic);
+            quizPointStatistic.setQuiz(quizExercise);
+        }
+
+        // make sure the pointers in the statistics are correct
+        quizExercise.recalculatePointCounters();
+
+        QuizExercise savedQuizExercise = super.save(quizExercise);
+
```

Comment:
Within a function, please do not mix var variable declarations with traditional variable declarations. Please stay consistent.

Reply:
change the var variable